lambda/audit.py

```diff
@@ -0,0 +1,319 @@
+                    "team": team,
+                    "repository": repo,
+                    "audit_id": audit_id,
+                }
+                LOG.info(event)
+                sns_repo_team_members = create_sns_message(
+                    {
+                        "action": "log_org_repo_team_members",
+                        "org": org,
+                        "repo": repo,
+                        "team": team,
+                        "audit_id": audit_id,
+                    }
+                )
+                publish_alert(sns_repo_team_members)
+        else:
+            LOG.error({"error": "Team not specified"})
+    return True
+
+
+def log_org_repos(message=None):
+    """ Audit github organization repositories """
+    org = os.environ.get("GITHUB_ORG")
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info({"action": "Audit organization org repos", "org": org})
+        repos = github_api.get_github_org_repositories(org)
+        event = {
+            "type": "OrganizationRepoCount",
+            "org": org,
+            "count": len(repos),
+            "audit_id": audit_id,
+        }
+        LOG.info(event)
+        for repo in repos:
+            event = {
+                "type": "OrganizationRepo",
+                "org": org,
+                "member": None,
+                "team": None,
+                "repository": repo,
+                "audit_id": audit_id,
+            }
+            LOG.info(event)
+            sns_repo_contributors = create_sns_message(
+                {
+                    "action": "log_org_repo_contributors",
+                    "org": org,
+                    "repo": repo,
+                    "audit_id": audit_id,
+                }
+            )
+            publish_alert(sns_repo_contributors)
+
+    return True
+
+
+def log_org_repo_contributors(message=None):
+    """ Audit github organization repository contributors """
+    org = os.environ.get("GITHUB_ORG")
+    repo = message.get("repo", None)
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info(
+            {
+                "action": "Audit organization org repo contributors",
+                "org": org,
+                "repo": repo,
+                "audit_id": audit_id,
+            }
+        )
+        repo_name = repo["name"]
+        if repo is not None:
+            members = github_api.get_github_org_repo_contributors(org, repo_name)
+            for member in members:
+                event = {
+                    "type": "OrganizationRepoContributor",
+                    "org": org,
+                    "member": member,
+                    "team": None,
+                    "repository": repo,
+                    "audit_id": audit_id,
+                }
+                LOG.info(event)
+        else:
+            LOG.error({"error": "Repo not specified"})
+    return True
+
+
+def log_org_repo_team_members(message=None):
```

Comment:
How is this data different to `log_org_team_repos`?

Reply:
org_team_repos = this repo can be accessed by members of this team (member = None)
org_repo_team_members = this repo can be accessed by this user because they are a member of this team (multiple entries for each team member) - means you don't have to join the data in splunk.

Reply:
Thanks for the explanation.